src/Voron/Impl/Journal/JournalReader.cs

```diff
@@ -54,15 +54,21 @@ public bool ReadOneTransactionToDataFile(StorageEnvironmentOptions options)
             if (_readAt4Kb >= _journalPagerNumberOfAllocated4Kb)
                 return false;
 
-            TransactionHeader* current;
-            if (TryReadAndValidateHeader(options, out current) == false)
+            if (TryReadAndValidateHeader(options, out TransactionHeader* current) == false)
             {
                 var lastValid4Kb = _readAt4Kb;
                 _readAt4Kb++;
+
                 while (_readAt4Kb < _journalPagerNumberOfAllocated4Kb)
                 {
                     if (TryReadAndValidateHeader(options, out current))
                     {
+                        if (CanIgnoreDataIntegrityErrorBecauseTxWasSynced(current, options))
```

Comment:
Here we need to see if the _next_ transaction is valid, (including the id), and only then agree to jump to it.
Because if the current tx is bad, we can't trust its length either

Reply:
Note that here `TryReadAndValidateHeader` returned `true` so we already found a valid transaction. `CanIgnoreDataIntegrityErrorBecauseTxWasSynced` call was to check that we're _still_ in this special mode (corrupted but already synced) and can continue the recovery.

Reply:
ok